### DISCUSSION TITLE
How does Vorta compare to Bup/Kup and BackInTime? Please add a comparison of features/functionality

### DISCUSSION THREAD

**Discussion by mYnDstrEAm** [borgbase/vorta#825] [General]
**Is your feature request related to a problem? Please describe.**
I couldn't find a clear and useful comparison of Borg/Vorta with [Bup/the Qt GUI Kup](https://github.com/bup/bup) and [BackInTime](https://github.com/bit-team/backintime/). These seem to be the three best FOSS backup software with a Qt GUI. Such a comparison would be useful for:
* users to find out whether Vorta is the adequate and/or best solution
* other developers to identify which Vorta features to add to their software
* developers of Vorta to identify which features other similar software has but Vorta lacks so that it can be added to make Vorta the best solution, allowing for all functionality of other software

**Describe the solution you'd like**
It would be good if a separate website and/or [the Vorta website](https://vorta.borgbase.com/) had a comparison of functionality/features of Vorta in direct comparison with (at least) these two other software. This comparison should probably be linked to either directly from "_Why is this great?_" in the readme or be findable at the link in "_Learn more on Vorta's website._"

Maybe this would be more appropriate at https://github.com/borgbase/docs.borgbase.com

* _BackInTime_ is great software but its development currently seems to be inactive. I suggested Vorta as an alternative to it here: https://github.com/bit-team/backintime/issues/1092#issuecomment-664042558 I'm mostly interested in how Vorta differs from it and which features BackInTime offers that Vorta currently does not have.
* _Kup_ is a Qt GUI for Bup and it seems to integrate nicely into KDE and, unlike Vorta, is in Debian's repos. However Bup isn't in the repos of latest Debian (only in older versions).

**Comment by m3nu** [comment]
Yeah, this could be added to the docs. But no problem to discuss it here and later do a PR there.

**Comment by Golddouble** [comment]
Here are some Ideas what could be included in this comparison:
(Maybe also LuckyBackup and "Deja Dup" could be included in this comparison. I know, that this are more sync programs, but nevertheless, actually they are also used to backup data)
1. How much time it takes to make a full backup of a sample of "folders and files".
2. How much time it takes to make something that feels like an incremental Backup of this sample of "folders and files".
3. How much time it takes to recover the latest backup of a sample of "folders and files".
4. How much time it takes to recover any particular "incremental backup" of a sample of "folders and files".
5. How much place a full backup takes with default settings of a sample of "folders and files".
6. How much place something takes that feels like a incremental backup. (of a sample of "folders and files")
7. Is it possible to brows through the files of the latest backup-archive?
8. Is it possible to browse through the files in any incremental backup-archive?
9. Can we recover single files or single folders from the latest backup or only the whole backup?
10. Can we recover single files or single folders from any incremental backup or only the whole backup?
11. How can I recover the whole home folder? Can I recover the whole home folder of the same system I am using at the moment? Or must I use a live system? Or something else?
12. Is there a function to verify my backups?
13. Is there an option that automatically verifies my backup after each incremental backup.
14. Can I verify only the latest incremental backup or do I have to verify always all backups of a repository
15. Can I encrypt my archive
16. Can I use a password
17. Can I delete any particular "incremental Backup" or only the latest?
18. Does it support differential Backups
19. When a file has changed since the latest backup, does then the backup program always save the whole file new or only parts of this file.
20. Is the backup program designed to backup "the whole home", "the root" or "only for your personal data, that are part of the home"?
21. Can we define exclusions rules
22. Is there an assistant for creating exclusion rules?
23. Is there a GUI for this program
24. Is the program actively supported (help)
25. Has it a good documentation?
26. For the recovery process: Can we define rules, if files that are already in the target should be deleted or kept. Already but only if it is newer. And other "sync rules".
27. Can I recover the content of a folder in my archive or particular files in a user defined target or are there any restrictions?
28. When recovering some single files: Is it easy to find them, because they are sorted in a logical manner. Or are they sorted in a mess?

**Comment by Golddouble** [comment]
This is a comparison between Borg, Restic, Arq 5, Duplicacy, Duplicati:
https://forum.duplicati.com/t/big-comparison-borg-vs-restic-vs-arq-5-vs-duplicacy-vs-duplicati/9952

**Comment by m3nu** [comment]
Cool. Borg is actually quite fast. I do wonder what's _Pcloud Virtual_ and if it runs Borg remotely. They should repeat it with BorgBase or another SSH-based backend.

**Comment by mat-m** [comment]
pCloud virtual is a fuse FS which caches the cloud data.

` fusermount -o rw,nosuid,nodev,auto_unmount,fsname=pCloud.fs,nonempty -- /home/me/pCloudDrive`

Edit: he did it on Windows, so does not count :-)

**Comment by mYnDstrEAm** [comment]
Thanks for all the useful constructive input here.  
From my experience with these two apps I'd _add_ these conclusions to comparisons:  

Vorta advantages
* deduplication
* compression
* under active development

BiT advantages
* allows backups of root files (which e.g. are needed for [system backups](https://unix.stackexchange.com/q/656393/233262)) (to do so one starts "Back in Time (root)" rather than "Back in Time" and configures the backups process once which then gets executed at regular intervals without having to enter the password again)
* allows backups of plain files without encryption and (common) forms of compression that don't allow quick file access (useful e.g. for syncing media centers)

I suggest Vorta incorporates any advantages other backup-software, especially BiT, have so that it becomes the best backup software and increases convenience & convergence rather than having to use multiple tools and developers inefficiently working on countless different software rather than mostly only a robust option-rich Vorta or forks/addons for it.

In specific enabling root backups is discussed [here](https://github.com/borgbase/vorta/discussions/801) and I'll probably ask about an unencrypted plain-file "synchronization"-mode backup for things like syncing media center files if that isn't already possible or planned or discussed elsewhere here. Edit: [created a feature request for it here](https://github.com/borgbase/vorta/issues/1161).
